### DISCUSSION TITLE
#Routing - Express Js Rest API

### DISCUSSION THREAD

**Discussion by TakayumAja** [deaafrizal/forum#15] [Troubleshooting Room] [answered]
Bagaiman membuat routing dengan parameter yang berbeda dalam rest api di Express Js.

```javascript
router.get("/:id", getDetailPackage);
router.get("/:name", getPackageLokasi);
```

Saat pengambilan di postman dengan localhost: localhost:3500/dikantor dan localhost:3500/1 tidak bisa.
Saat pengambilan req.params.id dan req.params.name tidak dapat memeroleh parameter req.params.name. Adakah caranya?

**Comment by deaafrizal** [comment, marked as answer]
tambahin bro, secara logika itu artinya apapun setelah "/" maka akan di detect sebagai data.

/123 => {data: 123}
/asd => {data: "asd"}

kalau pengen beda, kasih pembeda. 
/package/id/:id
/package/name/:name

lebih advance paramnya bisa dibuat lebih dinamis.
/:id.:name
ntar dapet {data: {id, name}} 
tinggal kasih validasi sesuai kebutuhan

**Reply by TakayumAja** [reply]
trimakasih bang, yang saya cari dari penggunaan param yang dinamisnya.

**Comment by djmaulana** [comment]
Punten bang dea mau nanya perihal API di nextjs saya lagi coba tutorial integrasi google sheet sama nextjs. Di localhost data berhasil di post dan get ke google sheet tapi pas saya coba deploy di vercel API nya tidak jalan. Route yang dipake getData [Ini api/getData](https://github.com/djmaulana/gSheetNextjs/blob/main/app/api/getData/route.ts), ini buat post nya [api/test untuk post data](https://github.com/djmaulana/gSheetNextjs/blob/main/app/api/test/route.ts), ini form buat get data yang mau di masukin ke google sheetnya [page.tsx](https://github.com/djmaulana/gSheetNextjs/blob/main/app/page.tsx). Hatur nuhun bang sateuacanaa

**Reply by deaafrizal** [reply]
mungkin harus di whitelist dulu di googlenya karena yang narik bukan localhost lagi bro
